### ISSUE TITLE
Parsing error for `timeout`

### ISSUE THREAD

**Issue by tariver** [hashicorp/terraform-provider-dns#298]
### Terraform CLI and Provider Versions

Greetings!
Stumbled into a bug with recent version. With the following configuration I get a syntax error:
```terraform
provider "dns" {
  update {
    server        = "x.x.x.x"
    key_name      = data.vault_generic_secret.bind_update_key.data["name"]
    key_algorithm = "hmac-sha512"
    key_secret    = data.vault_generic_secret.bind_update_key.data["secret"]
    timeout       = 5
    retries       = 5
  }
}
```

This is a version issue for certain, because if I add a version restriction
```terraform
terraform {
  required_providers {
    dns = {
      version = "<= 3.3.0"
    }
  }
}
```
everything runs fine.

Tested it out, version later than 3.3.0 (3.3.1  and 3.3.2) return the error.
I tried `5`, `5s` and `500ms` - all of them returned the same error.

### Terraform Configuration

```terraform
Terraform v1.0.11
on linux_amd64
+ provider registry.terraform.io/hashicorp/cloudinit v2.3.2
+ provider registry.terraform.io/hashicorp/consul v2.17.0
+ provider registry.terraform.io/hashicorp/dns v3.3.2
+ provider registry.terraform.io/hashicorp/vault v3.15.0
+ provider registry.terraform.io/hashicorp/vsphere v2.3.1
```

### Expected Behavior

Output of a plan

### Actual Behavior

A syntax error is returned
```
│ Error: Invalid timeout:
│ 
│   with provider["registry.terraform.io/hashicorp/dns"],
│   on provider.tf line 38, in provider "dns":
│   38: provider "dns" {
│ 
│ timeout cannot be parsed as an integer: strconv.Atoi: parsing "\"5\"": invalid syntax
```

### Steps to Reproduce

1. Have the following provider configuration:
```terraform
provider "dns" {
  update {
    server        = "x.x.x.x"
    key_name      = data.vault_generic_secret.bind_update_key.data["name"]
    key_algorithm = "hmac-sha512"
    key_secret    = data.vault_generic_secret.bind_update_key.data["secret"]
    timeout       = 5
    retries       = 5
  }
}
```
2. `terraform plan`

### How much impact is this issue causing?

Low

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bflad** [comment]
Hi @tariver 👋 Thank you for reporting this issue and sorry for the trouble. It does appear that there is an issue in the provider configuration handling introduced with version 3.3.0 and later. We'll ensure there is covering testing for this type of configuration and resolve the issue shortly. Thanks again.

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 3.3.3. Thanks again for reporting this.

**Comment by austinvalle** [comment]
Circling back on this issue, the fix for this has just been released in [`v3.4.0`](https://github.com/hashicorp/terraform-provider-dns/releases/tag/v3.4.0) with a couple other dependency updates, apologies for the delay!

It may take an hour or so to update in the registry cache. Thanks!
